src/firejail/main.c

```diff
@@ -2161,11 +2161,19 @@ int main(int argc, char **argv, char **envp) {
 		// hostname, etc
 		//*************************************
 		else if (strncmp(argv[i], "--name=", 7) == 0) {
+			int only_numbers = 1;
 			cfg.name = argv[i] + 7;
 			if (strlen(cfg.name) == 0) {
 				fprintf(stderr, "Error: please provide a name for sandbox\n");
 				return 1;
 			}
+			for (const char *name = cfg.name; *name && only_numbers; name++) {
+				only_numbers = isdigit(*name);
+			}
```

Comment:
It turns out this code snippet is wrong for two reasons. I've pushed it fixed

Reply:
@layderv [on Jan 15](https://github.com/netblue30/firejail/pull/5578#discussion_r1070539042):

> It turns out this code snippet is wrong for two reasons. I've pushed it fixed

Sorry, I hadn't tested it; now I see that `while (*c++) { ... }` doesn't make
sense.

Indeed, for it to work it would have had to be `do { ... } while (*++c);` or
`!isdigit(*c++)`.

Anyway, the fix looks good.

Reply:
Thank you for the review!